src/Drupal/DKANExtension/Context/WorkflowContext.php

```diff
@@ -30,23 +35,27 @@ public function transitionModerationState($named_entity, $state, $date = null) {
       throw new \Exception("State '$state' is not available. All possible states are [$possible_states].");
     }
 
-    $current_user = $this->getCurrentUser();
+    $current_user = $user = ($workflow_user) ? user_load_by_name($workflow_user) : $this->getCurrentUser();
```

Comment:
Yes, I'm changing the code to switch the user right before the moderation and set it back just after.